### ISSUE TITLE
[Feature Request] - How can we include the UK for Amazon, Royal Mail, DHL, etc ?

### ISSUE THREAD

**Issue by nytram-md** [moralmunky/Home-Assistant-Mail-And-Packages#262]
This looks a great HACS for Americans but it would be great if we could add our own delivery options also.

Im guessing the UK versions of Amazon and DHL would not be too much but if we could have Royal Mail and Parcelforce. That would be amazing.

I see someone else has already asked for Germany to be added

Thanks

Martyn

**Comment by firstof9** [comment]
Amazon UK is already supported.

In order to properly support other shippers, we'd need sample emails.

**Comment by nytram-md** [comment]
Hi, the actual email or a screenshot ?

Ill see if I have any

Thanks for the quick response

Martyn


Regards

Martyn

________________________________

**Comment by firstof9** [comment]
Yes the actual email, if you're using Gmail you can review [the steps here](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/issues/206#issuecomment-699272620) on how to download the email(s).

We'd need samples for "out for delivery" and "delivered", if applicable. 
Note even old emails will be fine, any personal info can be redacted, but I do ask that if tracking numbers are present please leave them intact so I can devise a regex for them.

**Comment by nytram-md** [comment]
Ill have a look and supply what information I can tomorrow.

Does amazon on the current build cover all countries or do i have to do anything as Im monitoring that now. ?

What about ebay as I buy a lot of stuff on there as this has a lot of the information for tracking ? I had a royal mail tracked package delivered today and everything was in ebay and it puts each stage in there and also in my hotmail calendar.


Regards

Martyn

________________________________

**Comment by firstof9** [comment]
> Does amazon on the current build cover all countries or do i have to do anything as Im monitoring that now. ?

It doesn't cover _all_ countries, but the ones listed in the const.py file.

> What about ebay as I buy a lot of stuff on there as this has a lot of the information for tracking ?

Since eBay isn't actually a delivery service like UPS, DHL, etc we don't utilize those.

**Comment by nytram-md** [comment]
Hi

Hotmail.com set up on smtp.live.com and port 993

Purchased 4 items from Amazon yesterday but nothing updated on the addon. It says its updating and I have the email in the inbox like I chose it to look.

Was there anything else I had to set up ?

Thanks

Martyn


Regards

Martyn

________________________________

**Comment by nytram-md** [comment]
Also trying Server

imap-mail.outlook.com<http://imap-mail.outlook.com>

993

Now


Regards

Martyn

________________________________

**Comment by firstof9** [comment]
According to outlook.com's help page these are the settings:
```
IMAP server name: outlook.office365.com
IMAP port: 993
```

**Comment by nytram-md** [comment]
Hi

Thanks for that. Ill try that but i put a post on Facebook this morning on a home assistant group and a few having issues in UK.

A guy I was speaking to said it was worded wrong for UK on the email search.

######

Okay, you stimulated me to debug it instead of my morning workout:)

There is indeed a bug (or rather lack of localisation) for amazon.co.uk<http://amazon.co.uk> email format. https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/a360c4b01a049f95f660374f95a1c66d456e4abe/custom_components/mail_and_packages/sensor.py#L744 expects email to contain "Will arrive:" and "Track your package:", while my emails have "Arriving:" and "Track your item(s):". Changing these strings make the integration work.

Martyn, if you are talking to author, please let him know about this. I can also prepare PR for this, but after my workout.

#####

Im not sure if that helps too. Ill dig an email out and have a look.

Martyn


Regards

Martyn

________________________________

**Comment by nytram-md** [comment]
________________________________

**Comment by firstof9** [comment]
> email to contain "Will arrive:" and "Track your package:", while my emails have "Arriving:" and "Track your item(s):". Changing these strings make the integration work.

This is part of the reason I ask for example emails 😉 
I then use those in the tests to see if the expected outcome occurs.

**Comment by nytram-md** [comment]
Yeah. Screen shot of a delivery on its way for me[Image]




Regards

Martyn

________________________________

**Comment by nytram-md** [comment]
This was the original email.

[Image]


Regards

Martyn

________________________________

**Comment by firstof9** [comment]
Sorry email images don't translate into github.
I'd need the actual email in order to properly setup tests.

**Comment by tomanderson2** [comment]
Heres a few amazon UK emails for your tests:
Out for Delivery: https://gist.github.com/tomanderson2/2c58b28dc4bc5de806267761c99dcbac
Delivered: https://gist.github.com/tomanderson2/335ada33a5ff449b99752cc5d420207c

and one more for luck (with multiple items if it makes a difference...)
Out for Delivery: https://gist.github.com/tomanderson2/08be4818503d5363ebc4b0dae42f1681
Delivered: https://gist.github.com/tomanderson2/b057c61296a84e112be69aa6e0fffbf6

**Comment by firstof9** [comment]
I'll get these Amazon UK ones sorted.
I have grabbed the gists.

**Comment by nytram-md** [comment]
updated to the latest version and its broken what I had working again.

My emails for tracking in the UK are "shipment-tracking@amazon.co.uk" but I cant see that in your new Amazon set up.

The search worked on "Track your Package" also

not sure if that helps.

**Comment by firstof9** [comment]
If you updated to 0.22 - Release, this isn't the 0.3.0 branch where these changes exist.
I'll get a 0.3.0-b0 released soon with these particular changes.

**Comment by moralmunky** [comment]
@nytram-md are you able to provide information similar to what is found in [supported shippers and requirements](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/wiki/Configuration-and-Email-Settings#supported-shippers-and-requirements) that provides a link to the service to sign up and details any required settings to begin receiving the correct email messages?

**Comment by clicky73** [comment]
Hi,
This is still not working for me for Amazon deliveries in the UK.

I think it may be because my emails are worded differently.
![image](https://user-images.githubusercontent.com/47227361/101981814-66dd2880-3c67-11eb-9100-a47ea333a21d.png)

Mine says Arriving: not will arrive:

Also when it is out for delivery it doesnt show anything
![image](https://user-images.githubusercontent.com/47227361/101981834-8e33f580-3c67-11eb-95be-26915db7032e.png)

I am running 0.3.0-b12.

What file do I need to change if at all to make it work?

Great integration BTW. Thanks

**Comment by tomanderson2** [comment]
@clicky73 I think this is still in progress (according to github anyway). It might be an idea to provide samples of your emails in eml format to help with development and testing, because screenshots don't quite cut it. Its easy to do, details in this post https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/issues/262#issuecomment-734483456

FWIW I anonymised my email address and tracking number when I shared mine, but there isn't much personal in there to worry about.

**Comment by clicky73** [comment]
[Amazon-UK-Email-Examples.zip](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/files/5683003/Amazon-UK-Email-Examples.zip)


Hope this helps.

There are 3 examples. Arriving date, delivering today and delivered.

**Comment by firstof9** [comment]
Give b13 a go. Should properly pick up your Amazon UK emails now.
